pkg/daemon/client.go

```diff
@@ -16,20 +18,26 @@ var (
 )
 
 type daemonClient struct {
-	client secretv1connect.SecretServiceClient
+	secretClient  secretv1connect.SecretServiceClient
+	controlClient controlv1connect.ControlServiceClient
 }
 
 func NewDaemonClient(ip string, port string) *daemonClient {
+	endpoint := fmt.Sprintf("http://%s:%s", ip, port)
 	return &daemonClient{
-		client: secretv1connect.NewSecretServiceClient(
+		secretClient: secretv1connect.NewSecretServiceClient(
 			http.DefaultClient,
-			fmt.Sprintf("http://%s:%s", ip, port),
+			endpoint,
+		),
+		controlClient: controlv1connect.NewControlServiceClient(
```

Comment:
I like the concept of having the services separated by their responsibility but I'm not sure if it is worth having two grpc connections in one client instance especially if we probably just use one at a time.
I we split the service, I would also have a different client wrappers.

In general I'm not sure if the `Shutdown` command is worth the overhead of having a separate client and grpc connection or if it is better to keep everything in a `DaemonService` for now and split it in the future if really needed.

Reply:
I am a little bit torn.

On the one hand, you are right. The new service adds overhead & complexity. It would only make sense to make this change now if we could already think about adding additional "control" functionality. I personally cannot think of any at the moment, apart from the `Shutdown`.

On the other hand, the separation of concerns and single responsibility patterns tell me that its probably not a good idea to mix the secret storage with control mechanisms. 

In general I would prefer to keep them separate, but I admit that this would only make sense in the future if we were to add more "control" functionality. I will create a single DaemonService instead.